File: dkb/dkb.go
package dkb

import (
	"rahul-rakshit/formatbankcsv/csv"
	"rahul-rakshit/formatbankcsv/utils"
)

type Formatter struct{}

func (f Formatter) ReadCsv(inputPath string) ([][]string, error) {
	delimiter := ";"
	skipLines := 4

	return csv.ReadCsv(inputPath, delimiter, skipLines)
}

func (f Formatter) Format(csvData [][]string, fromDate string, toDate string) ([][]string, error) {
	formatted, formatErr := formatDkb(csvData)
	if formatErr != nil {
		return [][]string{}, formatErr
	}

	filtered, filterErr := utils.FilterByDate(formatted, fromDate, toDate)
	if filterErr != nil {
		return [][]string{}, filterErr
	}

	return utils.SortByDate(filtered), nil
}

func (f Formatter) WriteCsv(data [][]string, outputPath string) error {
	delimiter := ","

	return csv.WriteCsv(data, outputPath, delimiter)
}

File: dkb/dkb.ts
import { readCsv, writeCsv } from "../csv/csv.ts";
import { filterByDate } from "../utils/filterByDate.ts";
import { formatDkb } from "./formatDkb.ts";

export class DkbFormatter {
  async readCsv(inputPath: string): Promise<string[][]> {
    const delimiter = ";";
    const skipLines = 4;
    return await readCsv(inputPath, delimiter, skipLines);
  }

  format(csvData: string[][], fromDate: string, toDate: string): string[][] {
    const formatted = formatDkb(csvData);
    if (fromDate || toDate) {
      return filterByDate(formatted, fromDate, toDate);
    }
    return formatted;
  }

  async writeCsv(data: string[][], outputPath: string): Promise<void> {
    const delimiter = ",";
    return await writeCsv(data, outputPath, delimiter);
  }
}
